test/test_spectral_ops.py

```diff
@@ -43,57 +47,162 @@ def method_fn(t):
     @skipCPUIfNoMkl
     @skipCUDAIfRocm
     @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
-    @precisionOverride({torch.complex64: 1e-4})
-    @dtypes(torch.complex64, torch.complex128)
-    def test_fft(self, device, dtype):
-        test_inputs = (torch.randn(67, device=device, dtype=dtype),
-                       torch.randn(4029, device=device, dtype=dtype))
-
-        def fn(t):
-            return torch.fft.fft(t)
-        scripted_fn = torch.jit.script(fn)
-
-        # TODO: revisit the following function if t.fft() becomes torch.fft.fft
-        # def method_fn(t):
-        #     return t.fft()
-        # scripted_method_fn = torch.jit.script(method_fn)
-
-        # TODO: revisit the following function if t.fft() becomes torch.fft.fft
-        # torch_fns = (torch.fft.fft, torch.Tensor.fft, scripted_fn, scripted_method_fn)
-        torch_fns = (torch.fft.fft, scripted_fn)
-
-        for input in test_inputs:
-            expected = np.fft.fft(input.cpu().numpy())
-            for fn in torch_fns:
-                actual = fn(input)
-                self.assertEqual(actual, expected, exact_dtype=(dtype is torch.complex128))
+    @precisionOverride({torch.complex64: 1e-4, torch.float: 1e-4})
+    @dtypes(torch.float, torch.double, torch.complex64, torch.complex128)
+    def test_fft_numpy(self, device, dtype):
```

Comment:
This test is really excellent.

Reply:
Overall these tests are very, very good. Is it interesting, after checking for consistency with our NumPy reference, to also check for stuff like `x = ifft(fft(x))`?

Also, are there any special cases or oddities we should be checking, like an fft formulation with -infs, infs, or NaNs?

Reply:
Added round-trip tests. Definitely makes sense so we can have a test that works independent of numpy.

>are there any special cases or oddities we should be checking, like an fft formulation with -infs, infs, or NaNs?

Not that I'm aware of. It's pretty much entirely adds and multiplications so should propagate nans and infs correctly by normal IEEE semantics.